lib/internal/quic/core.js

```diff
@@ -269,13 +270,27 @@ function onSessionClose(code) {
   session.destroy();
 }
 
-function onSessionHandshake(servername, alpn) {
+function onSessionHandshake(servername, alpn, cipher, cipherVersion) {
   debug('The session handshake is completed. ' +
         `Servername: ${servername}, ALPN Protocol: ${alpn}`);
   const session = this[owner_symbol];
   session[kState].servername = servername;
   session[kState].alpn = alpn;
-  process.nextTick(emit.bind(session, 'secure', servername, alpn));
+  session[kState].cipher = cipher;
+  session[kState].cipherVersion = cipherVersion;
+
+  const { minDHSize } = session[kState];
+  if (session instanceof QuicClientSession && typeof minDHSize === 'number') {
+    const ekeyinfo = session.ephemeralKeyInfo;
+    if (ekeyinfo.type === 'DH' && ekeyinfo.size < minDHSize) {
+      session.destroy(new ERR_TLS_DH_PARAM_SIZE(ekeyinfo.size));
+      return;
+    }
+  }
+
+  process.nextTick(emit.bind(session, 'secure',
+                             servername, alpn,
+                             session.cipher));
```

Comment:
Why are you metting on `nextTick`? Where `'secure'` not emitted before?

Reply:
Emitting on next tick is likely going to change. Just haven't done it yet